pandas/_libs/internals.pyx

```diff
@@ -890,17 +890,25 @@ cdef class BlockValuesRefs:
     """
     cdef:
         public list referenced_blocks
+        public int clear_counter
 
     def __cinit__(self, blk: Block | None = None) -> None:
         if blk is not None:
             self.referenced_blocks = [weakref.ref(blk)]
         else:
             self.referenced_blocks = []
+        self.clear_counter = 500  # set reasonably high
 
-    def _clear_dead_references(self) -> None:
-        self.referenced_blocks = [
-            ref for ref in self.referenced_blocks if ref() is not None
-        ]
+    def _clear_dead_references(self, force=False) -> None:
```

Comment:
would add some
comments here on why you are doing this / algo and reg the issue

Reply:
done